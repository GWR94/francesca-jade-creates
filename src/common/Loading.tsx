import React from "react";
import { CircularProgress, makeStyles } from "@material-ui/core";

interface LoadingProps {
  size?: number;
  color?: "inherit" | "primary" | "secondary";
  small?: boolean;
}

/**
 * A component which shows a loading spinner to the user to signal that there is
 * some loading going on.
 * @param {number} size - The size of the loading spinner
 * @param {string} color - The colour of the loading spinner
 */
const Loading: React.FC<LoadingProps> = ({
  size = 100,
  color = "inherit",
  small = false,
}): JSX.Element => {
  const useStyles = makeStyles({
    loading: {
      display: "flex",
      justifyContent: "center",
      alignItems: "center",
      width: "100%",
      height: "calc(100vh - 60px)",
    },
    loadingSmall: {
      display: "flex",
      justifyContent: "center",
      alignItems: "center",
      width: "100%",
      padding: 60,
    },
    spinner: {
      margin: "0 auto",
      display: "block",
    },
  });
  const classes = useStyles();
  return (
    <div className={small ? classes.loadingSmall : classes.loading}>
      <CircularProgress size={size} color={color} className={classes.spinner} />
    </div>
  );
};

export default Loading;
