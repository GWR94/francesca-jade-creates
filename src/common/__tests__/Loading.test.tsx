import React from "react";
import { Spinner } from "@blueprintjs/core";
import { shallow } from "enzyme";
import Loading from "../Loading";

describe("<Loading />", () => {
  it("should render correctly with no props", () => {
    const loading = shallow(<Loading />);
    expect(loading).toMatchSnapshot();
    expect(loading.find("div.loading__container").length).toBe(1);
    expect(loading.find(Spinner).length).toBe(1);
    expect(loading.find(Spinner).prop("size")).toBe(100);
    expect(loading.find(Spinner).prop("className")).toBe("loading__spinner");
  });

  it("should render smaller spinner based on size prop being 10", () => {
    const loadingSmall = shallow(<Loading size={10} />);
    expect(loadingSmall.find("div.loading__container").length).toBe(1);
    expect(loadingSmall.find(Spinner).length).toBe(1);
    expect(loadingSmall.find(Spinner).prop("size")).toBe(10);
    expect(loadingSmall.find(Spinner).prop("className")).toBe("loading__spinner");
  });

  it("should render larger spinner based on size prop being 200", () => {
    const loadingLarge = shallow(<Loading size={200} />);
    expect(loadingLarge.find("div.loading__container").length).toBe(1);
    expect(loadingLarge.find(Spinner).length).toBe(1);
    expect(loadingLarge.find(Spinner).prop("size")).toBe(200);
    expect(loadingLarge.find(Spinner).prop("className")).toBe("loading__spinner");
  });
});
