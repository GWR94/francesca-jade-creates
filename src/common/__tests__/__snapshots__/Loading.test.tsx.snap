// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Loading /> should render correctly with no props 1`] = `
<div
  className="loading__container"
>
  <Blueprint3.Spinner
    className="loading__spinner"
    size={100}
  />
</div>
`;
